src/pymovements/dataset/dataset_files.py

```diff
@@ -318,16 +272,75 @@ def load_gaze_file(
     """
     if custom_read_kwargs is None:
         custom_read_kwargs = {}
+    add_columns = {
+        key: fileinfo_row[key] for key in
+        [key for key in fileinfo_row.keys() if key != 'filepath']
+    }
 
     if filepath.suffix in {'.csv', '.txt', '.tsv'}:
         if preprocessed:
-            gaze_df = pl.read_csv(filepath)
+            gaze_df = from_csv(
+                filepath,
+                trial_columns=definition.trial_columns,
+                add_columns=add_columns,
+                column_dtypes=definition.filename_format_dtypes,
+            )
+
+            pixel_columns: list[str] = list(
+                set(GazeDataFrame.valid_pixel_position_columns) & set(gaze_df.frame.columns),
+            )
+            position_columns: list[str] = list(
+                set(GazeDataFrame.valid_position_columns) & set(gaze_df.frame.columns),
+            )
+            velocity_columns: list[str] = list(
+                set(GazeDataFrame.valid_velocity_columns) & set(gaze_df.frame.columns),
+            )
+            acceleration_columns: list[str] = list(
+                set(GazeDataFrame.valid_acceleration_columns) & set(gaze_df.frame.columns),
+            )
+
+            column_specifiers: list[list[str]] = []
+            if len(pixel_columns) > 0:
```

Comment:
I remember you telling me that this is quite hacky but let's leave it as is because I also don't know how to solve it more elegantly.

Reply:
k. no change